dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -1570,87 +1572,40 @@ private void setAccessiblePrograms(User user, EventSearchParams params) {
     }
   }
 
-  private String getOrgUnitSql(User user, EventSearchParams params, String ouTable) {
-    OrganisationUnitSelectionMode orgUnitSelectionMode = params.getOrgUnitSelectionMode();
-
-    if (orgUnitSelectionMode == null) {
-      if (params.getOrgUnit() != null) {
-        return getSelectedOrgUnitsPath(params, ouTable);
-      }
-
-      return getAccessibleOrgUnitsPath(params, user, ouTable);
-    }
-
-    return switch (orgUnitSelectionMode) {
-      case ALL -> null;
-      case CHILDREN -> getChildrenOrgUnitsPath(params, ouTable);
-      case DESCENDANTS -> getDescendantOrgUnitsPath(params, ouTable);
-      case CAPTURE -> getCaptureOrgUnitsPath(user, ouTable);
-      case SELECTED -> getSelectedOrgUnitsPath(params, ouTable);
-      default -> getAccessibleOrgUnitsPath(params, user, ouTable);
-    };
-  }
-
-  private String getChildrenOrgUnitsPath(EventSearchParams params, String ouTable) {
-    return ouTable
-        + "."
-        + PATH_LIKE
-        + " '"
-        + params.getOrgUnit().getPath()
-        + "%' "
-        + " and "
-        + ouTable
-        + "."
-        + "hierarchylevel = "
-        + (params.getOrgUnit().getLevel() + 1);
-  }
-
-  private String getSelectedOrgUnitsPath(EventSearchParams params, String ouTable) {
-    return ouTable + "." + PATH_EQ + " '" + params.getOrgUnit().getPath() + "' ";
-  }
-
-  private String getDescendantOrgUnitsPath(EventSearchParams params, String ouTable) {
-    return ouTable + "." + PATH_LIKE + " '" + params.getOrgUnit().getPath() + "%' ";
+  private String getChildrenOrgUnitsPath(List<OrganisationUnit> orgUnits, String ouTable) {
+    StringJoiner orgUnitSqlJoiner = new StringJoiner(" or ");
+
+    for (OrganisationUnit orgUnit : orgUnits) {
+      orgUnitSqlJoiner.add(
+          ouTable
+              + "."
+              + PATH_LIKE
+              + " '"
+              + orgUnit.getPath()
+              + "%' "
+              + " and "
+              + ouTable
+              + "."
+              + "hierarchylevel = "
+              + orgUnit.getLevel());
```

Comment:
do we not need to +1 anymore as we already got the children into `params.getAccessibleOrgUnits()`?

Reply:
Exactly. Also, the +1 was forcing us to get the children only, but we also want to get the parent (the org unit in the request).